### ISSUE TITLE
[Feature Request] search related

### ISSUE THREAD

**Issue by 4Magma** [dipu-bd/lightnovel-crawler#791]
Could you add a filter for search engines?

- I want to skip unwanted searches like from machine translater sites

Could you add an option to exit after sending selected url to stdout?

- I have a bash/python script to manage a list of novels
- currently copy url from terminal then exit

Could you add an option to check for update (new chapter) to a given url?

- I don't want to download a complete novel for a few new chapters

alternative:
I transmigrate from bash to python to manage my novels. Is there a way
to get above features by importing lncrawl and setting something before 
starting the app?

**Comment by dipu-bd** [comment]
### Could you add a filter for search engines?

Yes you can. Run with`lncrawl -x` or `lncrawl --sources`. It will show you a list of sources to select from. You can select only specific sources to search here.

### Could you add an option to exit after sending selected url to stdout?

Run using `lncrawl -l` or `lncrawl -ll` or `lncrawl -lll`. You will see more information, the more `l` you put here. Novel URL's are printed at level 2, or `lncrawl -ll`

### Could you add an option to check for update (new chapter) to a given url?

Once you have downloaded a novel, it will generate a folder containing chapter json files in it. As long as you don't delete this folder, the app will not redownload existing chapters again. Pass the `--ignore` option, i.e.  `lncrawl -i` to tell the app not to delete existing folder.

----------

To sum it up, run using `lncrawl -ll -i -x` and you will find all above features.

**Comment by 4Magma** [comment]
(**Could you add a filter for search engines?**)
- is there a permanent version of --sources, because each time to select would be tiring.
  (a config setting or env variable)
- want to restrict to novelfull.com, boxnovel.com and listnovel.com which showed the 
 fewest differences to original sites. (i create epubs for a tts reader and some constructs in 
 poorly grabbed texts are awful.)

(**Could you add an option to exit after sending selected url to stdout?**)
- Url is in line  "Choose a source to download? 1. ..." or in Debug line "Selected novel".
- To Stop at|before input of download folder i must still use Ctrl-C
- do you know a way to catch output from lncrawl to a env variable,
  i tried $(lncrawl ... 2>&1) (use bash on WSL2 from Win 10)

(**Could you add an option to check for update (new chapter) to a given url?**)
- i want to check if a novel has new chapters before downloading, 
  so i can prevent to start a process for none or too few chapters
- i already use the ignore option, but each time lncrawl walks through all chapters.
  - For novels with 1k+ chapters this process is boring.
  - i can't see if a chapter is only processed or also new downloaded
  - do you check for changes in old chapters? because i don't need updates of 
old chapters that i have already read
  - could you restrict processing of old chapters or at least the last n old chapters

**Extra**
Is it possible to create something like a changelog in a epub|book? Each update 
appends an entry which leads to the first new added chapter.
Each time i open a updated epub in my tts reader all my bookmarks are gone.
To find the last chapter i read i must note it externally.

**Comment by dipu-bd** [comment]
**(Could you add an option to check for update (new chapter) to a given url?)**

- The old chapters are not being processed. The the list of chapters are getting downloaded. Fetching this list is important to detect changes. When the app starts to download chapters, it checks whether a file exists. If file exists, it does not redownload the chapter again. You can also use `--last 50` to only download the last 50 chapters.

**etc.**

Most of the features you are requesting are not very useful for the general users. You can import `lncrawl` as a python module and do the necessary changes by yourself.


**Extra**

- You can use `--filename NAME` to set the output file name. If you generate different output file each time, you don't need changelog.

**Comment by dipu-bd** [comment]
I can work on these features:

- a filter for search engines to select specific ones by command line only
- add a `--dry-run` option, that will search and show the novel and its chapter list. but will not download anything.
- do not show any logs for checking chapters, but only for downloading.

**Comment by 4Magma** [comment]
_- a filter for search engines to select specific ones by command line only_

sounds good

_- add a --dry-run option, that will search and show the novel and its chapter list. but will not download anything._

sounds good

_- do not show any logs for checking chapters, but only for downloading._

the progress is showed by 
3 volumes and 258 chapters found
Downloading chapters |████████████████████████████████| 258/258
Downloaded 258 chapters

How about to change to
|████████████████████████████████| 258/258 {checked|downloaded}
n new chapters downloaded

or split in checked and downloaded
|██████████████████████████████     | 243/258 checked|present|existent
download 15 
|██                                                                              | 2/15

**Comment by dipu-bd** [comment]
Yeah, the `Downloaded X chapters` message is a bit confusing. I will fix that.

**Comment by dipu-bd** [comment]
You can now filter out which sources to search for novels, using a regex query. e.g: 

![image](https://user-images.githubusercontent.com/5158124/115977333-c134b100-a598-11eb-9d14-6816c0e3e947.png)

**Comment by 4Magma** [comment]
Could you add a command line option to skip the search selection and use all filtered?
e.g. -xx
